src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -322,56 +324,58 @@ public boolean isFullBlock(final IBlockState state)
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
 
-    /**
-     * Convert the BlockState into the correct metadata value.
-     *
-     * @deprecated (Remove this as soon as minecraft offers anything better).
-     */
     @NotNull
     @Override
-    @Deprecated
-    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
+    protected BlockStateContainer createBlockState()
     {
-        return state.withProperty(FACING, rot.rotate(state.getValue(FACING)));
+        return new BlockStateContainer(this, FACING, VARIANT);
     }
 
-    /**
-     * @deprecated (Remove this as soon as minecraft offers anything better).
-     */
-    @NotNull
     @Override
-    @Deprecated
-    public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mirrorIn)
+    public boolean hasTileEntity(final IBlockState state)
     {
-        return state.withRotation(mirrorIn.toRotation(state.getValue(FACING)));
+        return true;
     }
 
-    /**
-     * @deprecated (Remove this as soon as minecraft offers anything better).
-     */
     @Override
-    @Deprecated
-    public boolean isFullCube(final IBlockState state)
+    public TileEntity createTileEntity(final World world, final IBlockState state)
     {
-        return false;
+        return new TileEntityRack();
     }
 
     /**
-     * @deprecated (Remove this as soon as minecraft offers anything better).
+     * This returns a complete list of items dropped from this block.
+     *
+     * @param world   The current world
+     * @param pos     Block position in world
+     * @param state   Current state
+     * @param fortune Breakers fortune level
+     * @return A ArrayList containing all items this block drops
      */
     @Override
-    @Deprecated
-    public boolean isOpaqueCube(final IBlockState state)
+    public List<ItemStack> getDrops(final IBlockAccess world, final BlockPos pos, final IBlockState state, final int fortune)
     {
-        return false;
+        final List<ItemStack> drops = new ArrayList<>();
+
+        drops.add(new ItemStack(this, 1));
+
+        return drops;
     }
 
-    @NotNull
+    /**
+     * Called when a user uses the creative pick block button on this block
+     *
+     * @param state  the state.
+     * @param target the target.
+     * @param world  the world.
+     * @param pos    the position.
+     * @param player the player.
+     * @return the block pick result.
+     */
     @Override
-    @SideOnly(Side.CLIENT)
-    public BlockRenderLayer getBlockLayer()
+    public ItemStack getPickBlock(final IBlockState state, final RayTraceResult target, final World world, final BlockPos pos, final EntityPlayer player)
     {
-        return BlockRenderLayer.SOLID;
+        return new ItemStack(this, 1);
     }
 
     public enum EnumType implements IStringSerializable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)